hypothesis-python/src/hypothesis/core.py

```diff
@@ -300,10 +303,33 @@ def wrapped_test(*arguments, **kwargs):
         )
 
 
-def execute_explicit_examples(
-    test_runner, test, wrapped_test, settings, arguments, kwargs
-):
-    original_argspec = getfullargspec(test)
+class ArtificialDataForExample(ConjectureData):
+    """Dummy object that pretends to be a ConjectureData object for the purposes of
+    drawing arguments for @example. Provides just enough of the ConjectureData API
+    to allow the test to run. Does not support any sort of interactive drawing,
+    but that's fine because you can't access that when all of your arguments are
+    provided by @example.
+    """
+
+    def __init__(self, kwargs):
+        self.__draws = 0
+        self.__kwargs = kwargs
+
+        def draw_bytes(data, n):
+            raise NotImplementedError()  # pragma: no cover
+
+        super(ArtificialDataForExample, self).__init__(
+            max_length=0, draw_bytes=draw_bytes
+        )
+
+    def draw(self, strategy):
+        assert self.__draws == 0
+        self.__draws += 1
+        return (), self.__kwargs
```

Comment:
It would be useful to have a comment here that it works by imitating the tuples() strategy of args and kwargs which is set up by \@given, otherwise LGTM.